src/net/peer/connect/connect.rs

```diff
@@ -267,3 +260,24 @@ fn validate_connect_request<UID: Uid>(
     }
     Ok(())
 }
+
+/// Spawns p2p rendezvous connect task on the specified event loop.
+///
+/// Gets peer info from rendezvous relay channel and sends connected tcp stream to connection
+/// receiver.
+///
+/// # Returns
+///
+/// connection receiver
+pub fn start_rendezvous_connect(
+    handle: &Handle,
+    rendezvous_relay: UnboundedBiChannel<Bytes>,
+) -> oneshot::Receiver<Result<TcpStream, SingleConnectionError>> {
+    let (conn_tx, conn_rx) = oneshot::channel();
+    let start_conn = TcpStream::rendezvous_connect(rendezvous_relay, handle)
+        .map_err(|_| SingleConnectionError::RendezvousConnect)
+        .then(move |result| conn_tx.send(result))
+        .then(|_| Ok(()));
```

Comment:
Small nit: could this be replaced with a `map_err`? Just to make it clearer that there isn't an `Ok` value of some kind being thrown away.

Reply:
Well, that's the signature required by [Handle::spawn()](https://docs.rs/tokio-core/0.1.10/tokio_core/reactor/struct.Handle.html#method.spawn)

The only possible error in this case could be failure to send: https://docs.rs/futures/0.1.17/futures/sync/oneshot/struct.Sender.html#method.send
And this would only happen if the receiving part (someone who's interested in connection) got closed. In such case, they only way to handle this is do nothing. Hence, forget about the result.